README.md

```diff
@@ -209,7 +226,24 @@ setTimestamp(new Date().getTime())
 
 - **flush**()
 
-Flushes the current MetricsContext to the configured sink and resets all properties, dimensions and metric values. The namespace and default dimensions will be preserved across flushes. Timestamp will be preserved if set explicitly via `setTimestamp()`.
+Flushes the current MetricsContext to the configured sink and resets all properties and metric values. The namespace and default dimensions will be preserved across flushes. Custom dimensions are preserved by default, but this behavior can be changed by setting `logger.flushPreserveDimensions = false`. Timestamp will be preserved if set explicitly via `setTimestamp()`.
+
+Examples:
+
+```js
+logger.flush()  // custom and default dimensions will be preserved after each flush
+```
+
+```js
+logger.flushPreserveDimensions = false
+logger.flush()  // only default dimensions will be preserved after flush()
+```
+
+```js
+logger.flushPreserveDimensions = false
+logger.resetDimensions(false)
+logger.flush()  // default dimensions are disabled - no dimensions will be preserved after flush()
```

Comment:
@markkuhn why don't we simply pass `flushPreserveDimensions` as an input parameter of `flush()` like you did for `useDefault` in `setDimensions()` ?

Reply:
Setting `flushPreserveDimensions` is meant as additional config for the customer that they can set once instead of having to use it on every flush call.